### ISSUE TITLE
Bug? in skbio.Tree.tips(include_self=True)

### ISSUE THREAD

**Issue by amnona** [biocore/scikit-bio#1501]
When using the tree.tips(include_self=True) on a node which is a tip, we get an empty list
example:
from six import StringIO

from skbio import TreeNode

tree = TreeNode.read(StringIO("((a,b)c,(d,e)f);"))

print(len(list(tree.tips())))
4
z=list(tree.tips())[0]

print(len(list(z.tips())))
0
print(len(list(z.tips(include_self=True))))
0

**Comment by wasade** [comment]
Ah, good catch! `include_self` is not being [passed](https://github.com/biocore/scikit-bio/blob/master/skbio/tree/_tree.py#L1317) to `postorder`.

**Comment by bosborne** [comment]
It looks like include_self is being passed. I think the issue is more of a formal one. tips() works by calling is_tip(), which in turns asks if if the node has children (return not self.children). A tip will not have a child, by default. Should a node be allowed to be a child of itself, if include_self=True?

**Comment by nreeve17** [comment]
It looks like this issue was fixed in #1508

**Comment by nreeve17** [comment]
I tried this in ipython and could not reproduce the error. 

```python
from skbio import TreeNode

tree = TreeNode.read(StringIO("((a,b)c,(d,e)f);"))

print(len(list(tree.tips())))
4

z = list(tree.tips())[

print(len(list(z.tips())))
0

print(len(list(z.tips(include_self=True))))
1
```
